src/app/shared/services/cart.service.ts

```diff
@@ -260,7 +233,7 @@ export class CartService {
         size: instance.size,
         color,
         count: instance.inventory.map(r => r.count).reduce((x, y) => +x + +y, 0),
-        instances: [instance],
+        instances: item.instances.filter(r => r.product_color_id === color._id).map(r => Object.assign(r, {quantity: r.inventory.map(i => i.count).reduce((a, b) => a + b, 0)})),
         tags: [],
```

Comment:
If set instances to sum of stock in inventories, you cannot read instances frkm item in above code (line 230 in new version)